### DISCUSSION TITLE
Phet ThreeJS scenenode movement

### DISCUSSION THREAD

**Discussion by hiiambob89** [scenerystack/community#23] [Q&A]
Hey, we're trying to get the ThreeJS scenenode to move with everything while resizing. However, when we try to move the scene node with a vector when resizing it does nothing. We tried putting it in a rootnode, VBox, etc. but then trying to move that while resizing made it fly off of the screen. Is there a way to do this?

**Comment by jessegreenberg** [comment]
Hi @hiiambob89 - if your ThreeJS scene is contained in a scenery DOM Node, I would expect positioning it with a dot.Vector2 to work. Could you share a code example where you are setting the position of your ThreeJS scene from a resize?

**Reply by hiiambob89** [reply]
Here is an example:
`
window.addEventListener('resize', () => {

sceneNode.leftCenter = new phet.dot.Vector2(panelInitial.rightCenter.x,panelInitial.rightCenter.y)
}
`
we have also tried moving the node outside of the resize, putting it within another node, and moving that, but nothing worked.

**Reply by jessegreenberg** [reply]
Hey @hiiambob89 - that snippet looks right to me. What is `panelInitial`? Here is an example that positions a three.js scene at the center of the screen in case it helps.

Another thought - setters like `leftCenter` place the Node relative to its parent. If the parent of the sceneNode has some translation, that will impact where it is in the global coordinate frame and might explain why it flew off screen in a test.

<details>

```js
// Set up the scenery Display.
const rootNode = new phet.scenery.Node();
const display = new phet.scenery.Display( rootNode );
document.body.appendChild( display.domElement );

// set up a scene with three.js
const scene = new THREE.Scene();
const camera = new THREE.PerspectiveCamera( 75, window.innerWidth / window.innerHeight, 0.1, 1000 );
const renderer = new THREE.WebGLRenderer();
renderer.setSize( window.innerWidth / 4, window.innerHeight / 4 );

// Not necessary, scenery will append the child when you add it to the scene graph.
// document.body.appendChild( renderer.domElement );

// Wrap the three.js element in a scenery DOM node so it can be used with scenery.
const domNode = new phet.scenery.DOM( renderer.domElement );
rootNode.addChild( domNode );

// A basic Three.js scene
const geometry = new THREE.BoxGeometry( 1, 1, 1 );
const material = new THREE.MeshBasicMaterial( { color: 0x00ff00 } );
const cube = new THREE.Mesh( geometry, material );
scene.add( cube );
camera.position.z = 5;

const animateCube = () => {
  cube.rotation.x += 0.01;
  cube.rotation.y += 0.01;
}

// Update the scenery display and Three.js every animation frame
display.updateOnRequestAnimationFrame( () => {

  // Also request a render from the three.js scene every animation frame.
  animateCube();
  renderer.render( scene, camera );
} );

// Resize the display to always fill the window
display.resizeOnWindowResize();

// Manage layout
const resizeListener = () => {
  domNode.center = new phet.dot.Vector2( window.innerWidth / 2, window.innerHeight / 2 );
};

window.addEventListener( 'resize', () => {
  resizeListener();
} );
resizeListener();
```
</details>
